### ISSUE TITLE
did something change with list-item-indent rule?

### ISSUE THREAD

**Issue by niftylettuce** [remarkjs/remark-lint#213]
It wasn't until today that I started getting these warnings, and the `remark` cli fix flag isn't working on them either.

```sh
npx remark . -qfo
ASSETS.md
   14:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   15:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   16:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   17:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   18:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   19:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   20:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   21:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   22:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   23:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   24:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   25:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   26:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   27:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint

DEVELOPER-ONBOARDING.md
   26:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
   27:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
   28:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  190:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  191:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  192:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  193:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  194:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  195:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  196:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint

README.md
   98:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  133:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  140:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  148:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  183:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  189:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  207:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  213:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  219:4  warning  Incorrect list-item indent: add 1 space   list-item-indent  remark-lint
  249:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  250:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  251:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  288:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  289:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  290:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint

app/emails/README.md
   22:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   23:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   24:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   25:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   30:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   35:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   36:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   37:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   44:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   45:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   46:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   47:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   52:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   53:5  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   54:5  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   55:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   56:5  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   57:5  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   58:5  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint

⚠ 58 warnings
```

**Comment by ChristianMurphy** [comment]
@niftylettuce A few follow up questons:
1. what version of `remark` is being used?
2. what version of `remark-lint` is being used?
3. what version of `remark-lint-list-item-indent` is being used?
4. Did you update your `package.json` recently? Or are you using a version range?
5. Could you provide a sample of markdown that is warning?

Thanks! :bow:

**Comment by niftylettuce** [comment]
remark-cli v6.0.1
remark-preset-github v0.0.13
...remark-lint-list-item-indent v1.0.3

**Comment by niftylettuce** [comment]
remark-lint@6.0.0

**Comment by wooorm** [comment]
https://github.com/remarkjs/remark-lint/commits/master/packages/remark-lint-list-item-indent does not show anything of significance that changed.

**Comment by niftylettuce** [comment]
OK so Yarn is the culprit.  Here's my `yarn.lock` file.  When I install with just npm it works fine, and no warning messages.  I didn't see any version issues, but maybe either of you could lend a hand?

<https://gist.github.com/niftylettuce/3fc34cca468e3a6c0ad9e3dff07bdd89>

**Comment by ChristianMurphy** [comment]
It looks like there are two different versions of the `remark-lint-list-item-indent`
https://gist.github.com/niftylettuce/3fc34cca468e3a6c0ad9e3dff07bdd89#file-yarn-lock-L15616-L15636

which might cause an issue.

Does running `yarn upgrade` then running the script resolve the issue?

**Comment by niftylettuce** [comment]
No, it doesn't

**Comment by ChristianMurphy** [comment]
Instead of using `npx` try using `yarn dlx` https://yarnpkg.github.io/berry/cli/dlx

**Comment by niftylettuce** [comment]
it yields same error, as does `node_modules/.bin/remark . -qfo`

**Comment by ChristianMurphy** [comment]
From the context provided, the source of the issue is still unclear, but appears to be an issue with `yarn` rather than `remark`.
Can you share the repository where the issue is happening? Or a minimal example repo/gist?

**Comment by niftylettuce** [comment]
@ChristianMurphy 	here you go: https://github.com/niftylettuce/remark-lint-213-reproduce-test

Simply `yarn` and then `npx remark . -qfo`, it yields this error:

```sh
npx remark . -qfo
TEST.md
  13:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  14:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  15:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint

⚠ 3 warnings
```

**Comment by arcanis** [comment]
> Instead of using `npx`, try using `yarn dlx` https://yarnpkg.github.io/berry/cli/dlx
> (see [yarnpkg/yarn#3937](https://github.com/yarnpkg/yarn/issues/3937) for context)

Yarn dlx is v2-exclusive (still in development), it won't work with the v1.x.

Regarding the general issue, you have two different versions of `remark-lint-list-item-indent` because two packages depend on it via different ranges. Yarn doesn't provide any guarantee that they will use the same version, even if intuitively you'd think it would (and the same is true for npm - if it works, it's purely by luck).

**Comment by wooorm** [comment]
@niftylettuce The problem seems to be that `remark-preset-github` locks its [`remark-lint-list-item-indent`] dependency (https://gist.github.com/niftylettuce/3fc34cca468e3a6c0ad9e3dff07bdd89#file-yarn-lock-L16005). You should unlock that dependency.

**Comment by niftylettuce** [comment]
You're the best @wooorm.  Completely forgot and didn't see easily that the version was locked, it was from way back when I had that insanely annoying yarn cache issue.

I've released a new version of `remark-preset-github` and all is good now 🎉 	🎉 	

Ref: https://github.com/niftylettuce/remark-preset-github/commit/2b8bd840ba7bded2dedf08d5c90b1a678b0ee6e6#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R26

**Comment by wooorm** [comment]
Sweet, no problem ✨
